tests/test_pulse.py

```diff
@@ -68,3 +68,199 @@ def test_field_cos_with_chirp(make_pulse):
     # chirp breaks the even symmetry of the cos function
     assert cos[0] != cos[-1]
     assert cos[1] != cos[-2]
+
+
+@pytest.mark.parametrize("envelope_type", ENVELOPE_TYPES)
+def test_field_envelope(make_pulse, envelope_type):
+    fwhm = 15.0
+    t0 = 0
+    pulse = make_pulse(fwhm=fwhm, t0=t0, envelope_type=envelope_type)
+
+    t = np.array([-64, -16, -4.0, 0.0, 4.0, 16, 64])
+
+    envelope = pulse.calc_field_envelope(t)
+
+    s = snapshot(
+        {
+            "gauss": {
+                0: 1.0960549698939707e-11,
+                1: 0.20653303211845794,
+                2: 0.9061223550383886,
+                3: 1.0,
+                4: 0.9061223550383886,
+                5: 0.20653303211845794,
+                6: 1.0960549698939707e-11,
+            },
+            "lorentz": {
+                0: 0.03209025202400356,
+                1: 0.346604598571491,
+                2: 0.8945978172388421,
+                3: 1.0,
+                4: 0.8945978172388421,
+                5: 0.346604598571491,
+                6: 0.03209025202400356,
+            },
+            "sech": {
+                0: 0.001083121940752962,
+                1: 0.2981611144195047,
+                2: 0.8988518704633442,
+                3: 1.0,
+                4: 0.8988518704633442,
+                5: 0.2981611144195047,
+                6: 0.001083121940752962,
+            },
+            "sin": {
+                0: 0.0,
+                1: 0.0,
+                2: 0.9135454576426009,
+                3: 1.0,
+                4: 0.913545457642601,
+                5: 0.0,
+                6: 0.0,
+            },
+            "sin2": {
+                0: 0.0,
+                1: 0.11811460943703252,
+                2: 0.90982893698841,
+                3: 1.0,
+                4: 0.90982893698841,
+                5: 0.11811460943703297,
+                6: 0.0,
+            },
+        }
+    )
+    s_tmin = snapshot(
+        {
+            "gauss": -36.0,
```

Comment:
@JanosJiri please double check that these values make sense. 

For example it's quite nice to see how much "wider" the lorentzian pulse is.

Reply:
I went through most of the data and it makes sense. I will change the `envelopes_analysis` and we will see if it still fulfils the FWHM conditions.